salt/utils/files.py

```diff
@@ -785,6 +785,20 @@ def backup_minion(path, bkroot):
         os.chmod(bkpath, fstat.st_mode)
 
 
+def case_sensitive_filesystem(path=None):
+    """
+    Detect case sensitivity on a system.
+
+    Returns:
+        bool: Flag to indicate case sensitivity
+
+    .. versionadded:: 3004
+
+    """
+    with tempfile.NamedTemporaryFile(prefix="TmP", dir=path, delete=True) as tmp_file:
+        return os.path.exists(tmp_file.name.lower())
```

Comment:
This would return False on Linux, but true on Windows. Windows is NOT case sensitive... the naming here is misleading.